MayssaraOmar/assignment3/Word search/src/test_dictionary.cpp

```diff
@@ -0,0 +1,93 @@
+#include "test_dictionary.h"
+#include "test.h"
+#include "exception"
+
+dictionary test_dictionary::dic;
+
+void test_dictionary::run_all_tests() {
```

Comment:
One problem with the current test setup is that all tests share the same dictionary, which means the order you run the tests has an effect on the result.
An alternative would be to make the test class have non-static methods and for each test you create an instance.

It would then look something like:

```c++
void test_dictionary::run_all_tests() {
  new test_directionary().insert_oneword();
  new test_directionary().insert_vector();
  ...
}
```